src/Microsoft.AspNet.OData/Routing/ODataPathRouteConstraint.cs

```diff
@@ -47,10 +47,31 @@ public virtual bool Match(HttpRequestMessage request, IHttpRoute route, string p
                 object oDataPathValue;
                 if (values.TryGetValue(ODataRouteConstants.ODataPath, out oDataPathValue))
                 {
+                    string odataPath = oDataPathValue as string;
+                    Func<IServiceProvider> requestContainerFactory = () =>
+                    {
+                        // Delegate is reused. We need to ensure only one request container is created
+                        if (request.Properties.ContainsKey(Extensions.HttpRequestMessageExtensions.RequestContainerKey))
```

Comment:
Might want to consider using a `TryGetValue` instead

Reply:
Fixed